src/clients/errors.rs

```diff
@@ -69,6 +69,23 @@ impl From<reqwest::Error> for Error {
     }
 }
 
+impl From<reqwest_middleware::Error> for Error {
```

Comment:
Identical to handling of `reqwest::Error`, except for the thin middlware wrapper crate `reqwest_middleware`